Different way of handling Media Uploads

wartman opened the pull request:
I've been using Webcomic for years now, and it's pretty great! The only complaint I have is the way media uploads are handled -- having to close the modal WITHOUT hitting 'Insert Into Post" feels weird. I've tried using `wp.media.view.MediaFrame.Select` here, and I think it works pretty well. At the very least clicking `Add Media` now does what you expect it to.

Thoughts? There might be something I'm missing about the approach you chose.

mgsisk commented:
Thanks for the pull request @wartman! I'm glad you like Webcomic, and I agree; media handling in Webcomic 4 is… really stupid. Webcomic 5 will address this, but it'd be nice to correct this in v4 since v5 is still on a "when it's done" availability schedule and the v5 solution to this – completely changing how comic media and posts are associated – isn't practical to port backwards into v4.

I'm glad you've taken a stab at this and I like where it's headed, but I noticed a few issues after a quick usability review (I haven't dived into the code yet):

- It looks like there's no way to detach or sort media items from the metabox or the media popup.
- The metabox preview appears to be broken. The Add Media button *always* appends media to the post, but when new media items are added the metabox only ever shows the last-selected media item until the page is reloaded. This makes it look like the selected media item has replaced the existing comic media, or that only the last-uploaded media item will be attached when several media items are uploaded at once.
- I'm not sure if this is an actual issue or just how the media popup works, but the dates dropdown shows dates for all uploaded media, even though the library itself is limited to media associated with the current post. The Media Library tab is kind of redundant here, to be honest, but – like those extraneous dates – I'm not sure if there's any good way to remove it.

wartman commented:
Awesome, glad to hear that you're working on the next iteration!

The code **is** pretty rough right now, but I'll take a stab at cleaning things up and addressing your points. The metabox preview will be a super simple fix, although I'm not sure about the other ones -- I'll have to go digging through the `wp.media` code again.

wartman commented:
You can now detach existing images from a post, and the library modal displays any unattached images that already exist on your site.

I'm sure there is a more elegant way of doing this, but I honestly don't know the ins-and-outs of the `wp.media` code well enough to do it. The only big issue now (due to this) is that the library view never updates, which can be a bit confusing. Still, I think it's an improvement.

mgsisk commented:
Thanks for your work @wartman. Just wanted to let you know I haven't forgotten this, will take a look soon!

wartman commented:
No worries, just happy to contribute!
